src/command_modules/azure-cli-cdn/azure/cli/command_modules/cdn/_params.py

```diff
@@ -0,0 +1,98 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import argparse
+
+from azure.mgmt.cdn.models import (QueryStringCachingBehavior, SkuName)
+
+from azure.cli.core.commands import (register_cli_argument, CliArgumentType)
+from azure.cli.core.commands.parameters import enum_choice_list, three_state_flag, tags_type
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from ._validators import validate_origin
+
+
+class OriginType(argparse._AppendAction):
+    def __call__(self, parser, namespace, values, option_string=None):
+        from azure.mgmt.cdn.models import DeepCreatedOrigin
+
+        if not (1 <= len(values) <= 3):
+            msg = '%s takes 1, 2 or 3 values, %d given'
+            raise argparse.ArgumentError(self, msg % (option_string, len(values)))
+
+        deep_created_origin = DeepCreatedOrigin('origin', values[0], http_port=80, https_port=443)
+        if len(values) > 1:
+            deep_created_origin.http_port = int(values[2])
+        if len(values) > 2:
+            deep_created_origin.https_port = int(values[3])
+        super(OriginType, self).__call__(parser, namespace, deep_created_origin, option_string)
+
+name_arg_type = CliArgumentType(options_list=('--name', '-n'), metavar='NAME')
+
+register_cli_argument('cdn', 'name', name_arg_type, id_part='name')
+register_cli_argument('cdn', 'tags', tags_type)
+
+register_cli_argument('cdn profile create',
+                      'sku',
+                      **enum_choice_list([item.value for item in SkuName.__members__.values()]))
+register_cli_argument('cdn profile create', 'location',
+                      validator=get_default_location_from_resource_group)
+register_cli_argument('cdn profile', 'profile_name', name_arg_type, id_part='name')
+
+register_cli_argument('cdn custom-domain create', 'location',
+                      validator=get_default_location_from_resource_group)
+
+
+register_cli_argument('cdn endpoint', 'endpoint_name', name_arg_type, id_part='name')
+
+# Endpoint Create #
+
+cdn_endpoint = 'cdn endpoint'
+register_cli_argument(cdn_endpoint, 'location',
+                      validator=get_default_location_from_resource_group)
+register_cli_argument(cdn_endpoint,
+                      'origins',
+                      options_list='--origin',
+                      nargs='+',
+                      action=OriginType,
+                      validator=validate_origin,
+                      help='Endpoint origin specified by the following space delimited 3 \
```

Comment:
It was a convention we established many, many moons ago for the commands.py and params.py file.  I'm not saying I'd block the PR for it... but it is inconsistent with the rest of the CLI's params files.